File: packages/node/src/utils/daemon-config.ts
import { getConfiguredPort } from './port.js';

/** Base URL for CLI / thin MCP to reach daemon (loopback, port from `PORT`). */
export function getDaemonHttpBaseUrl(): string {
  return `http://127.0.0.1:${getConfiguredPort()}`;
}

/** Optional shared secret; if set, HTTP requests must send `Authorization: Bearer <token>`. */
export function getDaemonAuthToken(): string | undefined {
  const t = process.env.DAEMON_AUTH_TOKEN?.trim();
  return t || undefined;
}

File: packages/python/src/lionscraper/utils/daemon_config.py
from __future__ import annotations

import os

from lionscraper.utils.port import get_configured_port


def get_daemon_http_base_url() -> str:
    return f"http://127.0.0.1:{get_configured_port()}"


def get_daemon_auth_token() -> str | None:
    t = os.environ.get("TOKEN", "").strip()
    return t or None
